### PR TITLE
CI job to verify ABI compatibility

### PR DESCRIPTION
**Pull request by praveek**
Added CI task to verify API/ABI changes. 

For builds to success after any public API changes, run `make api-dump` and commit the generated `android-core-library.api` file as part of your PR.

### CODE REVIEW

**Review by prudrabhat — APPROVED** [review]

File: code/android-core-library/api/android-core-library.api [hunk 1]
```diff
@@ -0,0 +1,1075 @@
...
+	public static final field CALLBACK_NULL Lcom/adobe/marketing/mobile/AdobeError;
+	public static final field CALLBACK_TIMEOUT Lcom/adobe/marketing/mobile/AdobeError;
+	public static final field EXTENSION_NOT_INITIALIZED Lcom/adobe/marketing/mobile/AdobeError;
+	public static final field UNEXPECTED_ERROR Lcom/adobe/marketing/mobile/AdobeError;
+	protected fun <init> (Ljava/lang/String;I)V
+	public fun getErrorCode ()I
+	public fun getErrorName ()Ljava/lang/String;
+}
+
+public final class com/adobe/marketing/mobile/BuildConfig {
```
Comment (yangyansong-adbe) [inline code/android-core-library/api/android-core-library.api:19]: Gradle generated the `BuildConfig` class, can we remove it here?

File: code/android-core-library/api/android-core-library.api [hunk 2]
```diff
@@ -0,0 +1,1075 @@
...
+	public fun loadFromCache (Ljava/lang/String;)Lcom/adobe/marketing/mobile/launch/rulesengine/download/RulesLoadResult;
+	public fun loadFromUrl (Ljava/lang/String;Lcom/adobe/marketing/mobile/AdobeCallback;)V
+}
+
+public final class com/adobe/marketing/mobile/launch/rulesengine/json/JSONRulesParser {
+	public static final field INSTANCE Lcom/adobe/marketing/mobile/launch/rulesengine/json/JSONRulesParser;
+	public static final fun parse (Ljava/lang/String;Lcom/adobe/marketing/mobile/ExtensionApi;)Ljava/util/List;
+}
+
+public class com/adobe/marketing/mobile/rulesengine/ComparisonExpression : com/adobe/marketing/mobile/rulesengine/Evaluable {
```
Comment (yangyansong-adbe) [inline code/android-core-library/api/android-core-library.api:397]: The classes under `com/adobe/marketing/mobile/rulesengine` package could be removed.

**Review by yangyansong-adbe — APPROVED** [review]
LGTM, only some small comments

File: code/android-core-library/api/android-core-library.api (hunk 2 above)
Comment (praveek) [reply code/android-core-library/api/android-core-library.api:397]: Let us move into internal subpackage if we don't wish to make it public.

**Pull request merged**